test/client/connection.js

```diff
@@ -142,11 +142,18 @@ describe('client connection', function() {
       expect(connection.state).equal('connecting');
     });
 
-    it('initial connection.state is connecting, if socket.readyState is OPEN', function() {
-      // https://html.spec.whatwg.org/multipage/web-sockets.html#dom-websocket-open
-      var socket = {readyState: 1};
+    it('initial connection.state is connecting and init handshake, if socket.readyState is OPEN', function() {
+      var send = false;
+      var socket = {
+        readyState: 1,
+        send: function(msg) {
+          expect(JSON.parse(msg).a).equal('hs');
+          send = true;
```

Comment:
Setting flags like this feels a little weird. Can we please use a spy instead? Like:

```javascript
var socket = {
  readyState: 1,
  send: sinon.spy(),
};

expect(socket.send.calledOnce).to.be.true;
const message = JSON.parse(socket.send.getCall(0).args[0]);
expect(message.a).to.equal('hs');
```

Reply:
fixed

Reply:
🙌 thanks!